iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,37 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  AllTypesMatch<["value", "result"]>
+  ]> {
+  let summary = [{returns the (value + (alignment - 1)) & ~(alignment - 1)}];
+  let description = [{
+    Aligns |value| up to the given power-of-two |alignment| if required.
+    https://en.wikipedia.org/wiki/Data_structure_alignment#Computing_padding
+
+    Usage:
+      %0 = util.align $value, $alignment : index
+  }];
+
+  let arguments = (ins
+    SignlessIntegerLike:$value,
+    Index:$alignment
+  );
+
+  let results = (outs
+    SignlessIntegerLike:$result
+  );
```

Comment:
@benvanik we weren't too sure about the types here. How do these look to you? If this op remains in IREE's util dialect (and doesn't move upstream to `arith`), we could constrain the types more.